### ISSUE TITLE
Hanging at Fetching Nest devices.

### ISSUE THREAD

**Issue by djfatal** [KraigM/homebridge-nest#125]
I started having trouble with the new API tonight.  When I run homebridge it will hang at Fetching Nest devices.  If I remove token from my config.json it will work in Legacy API.

**Comment by dvd2000k** [comment]
Same issue here...

**Comment by PencilDH** [comment]
Ditto -> stuck on fetching devices , crashed last night 

```
/usr/local/lib/node_modules/homebridge-nest/node_modules/firebase/node_modules/faye-websocket/node_modules/websocket-driver/node_modules/websocket-extensions/lib/websocket_extensions.js:133
    this._pipeline.processIncomingMessage(message, callback, context);
                  ^

TypeError: Cannot read property 'processIncomingMessage' of undefined
    at instance.processIncomingMessage (/usr/local/lib/node_modules/homebridge-nest/node_modules/firebase/node_modules/faye-websocket/node_modules/websocket-driver/node_modules/websocket-extensions/lib/websocket_extensions.js:133:19)

```

**Comment by PencilDH** [comment]
#109 #106 ?? Seems like it's not crashing but just hanging now?

**Comment by julianrv** [comment]
For me it just hangs but it prevents homebridge from starting, perhaps it would be a good idea to implement some timeout to avoid this situation

**Comment by fafoulon** [comment]
Same problem for me, was crashing in the night, and now it's hanging

**Comment by luisiam** [comment]
Same issue. Now I have to remove the plugin temporarily

**Comment by PencilDH** [comment]
Did anyone try to fall back to legacy API of now? I didn't have time to check it out yet?

**Comment by Danno665** [comment]
Same issue here.
Resulting in installing everything from the ground up.... :)

I wanted to install a new configuration of Homebridge anyway.
Strange thing is, the beginning of the configuration with creating a token with pincode still works.

I guess it has to do something with the status of the Nest itself.
App of Nest is working just correctly.

**Comment by sleepyarmistice** [comment]
Running home bridge on an rpi, tried to go to legacy as described above to no effect.

**Comment by PencilDH** [comment]
@sleepyarmistice I run rpi and it works .... did you thing say Legay API on terminal, if not you didn't fall back to legacy?

**Comment by sleepyarmistice** [comment]
Oh, geez, no. I'll try it when I get home and it will most likely work.

**Comment by neutrl** [comment]
Ditto, same issue here.  Even attempted to re-create the app on the Nest developer portal. 

Legacy authorization still works fine.

**Comment by d3bing** [comment]
Same here...

**Comment by gerarddvb** [comment]
Same here, broke my setup.

**Comment by joe248** [comment]
Same issue starting last night.  I don't think Homebridge itself is hanging because I'm still seeing log messages from my other devices, but trying to use the new Nest API causes all devices in the iOS Home app to show "No Response".  Using the legacy auth for homebridge-nest works okay.

**Comment by Slepperpon** [comment]
Same problem here removed nest and all other platforms work just fine.

**Comment by djfatal** [comment]
@PencilDH Yes the Legacy API is still working.  Can't figure out why it hangs on the new API.

**Comment by safzal** [comment]
I have the same problem, and after looking into this its look like https://deverloper-api.nest.com on the new API response: it's giving an error: 

**Internal Server Error**
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

**Comment by kcharwood** [comment]
Same issue

**Comment by starlessblack** [comment]
Reverted to legacy mode for the time being. Any ideas on what the long-term fix.

**Comment by mgawley4** [comment]
I'm tracking this thread as well as I have the same problems.

**Comment by gizbod** [comment]
Same issue, following

**Comment by starlessblack** [comment]
Anyone try regenerating the client token of the homebridge nest plugin?

**Comment by neutrl** [comment]
> Anyone try regenerating the client token of the homebridge nest plugin?

I have, didn't seem to change anything.  I just fell back to legacy for now.

**Comment by djfatal** [comment]
I was actually using the camera's motion and smoke and co sensor with the new API. It was nice that when motion was detected I would get a notification with a camera image as well. But in legacy the only thing that works is the thermostat.

**Comment by safzal** [comment]
it looks like this issue has been solved it working now...

**Comment by d3bing** [comment]
yup, revert to new API and back on track... guess it was Nest

**Comment by djfatal** [comment]
Good to know, will try it when I get home tonight!  Thanks @safzal

**Comment by fafoulon** [comment]
I activated it again, and it's also working for me now

**Comment by HappySpaceInvader** [comment]
Getting the same issue - legacy API works fine.  Completely recreated a new product at developer.nest.com, generates token fine, but hands when trying to connect.

Is there a fix for this?

**Comment by Krocko** [comment]
Same for me. Has anyone a solution?

**Comment by chrisjshull** [comment]
OK, a couple of things I'd try:

1. use my fork, it has a lot of fixes from people merged in: `npm install -g "https://github.com/chrisjshull/homebridge-nest"`
1. run homebridge with debug logging: `DEBUG=* homebridge -D`
1. I don't think this is the issue people are seeing, but it should be noted that the NEST API has a rate limit that will just reject commands for a while if you trigger it.

Let me know how that goes, and share whatever logging you get.

**Comment by HappySpaceInvader** [comment]
Somewhat embarassingly, I think it was option 3. from your list in my case.  I left it a couple of days, and then tried again with my token, and homebridge started up fine, adding the Nest device.

I had had some spurious 2-factor auth text messages recently that seemed to indicate someone had tried to access my account which perhaps caused it to become locked.

I may try your fork out anyway, as I feel the plugin should handle this situation a little more gracefully.
